cmake/modules/platform/arch/riscv.cmake

```diff
@@ -0,0 +1,48 @@
+###############################################################################
+# Copyright (c) 2019 IBM Corp. and others
+#
+# This program and the accompanying materials are made available under
+# the terms of the Eclipse Public License 2.0 which accompanies this
+# distribution and is available at http://eclipse.org/legal/epl-2.0
+# or the Apache License, Version 2.0 which accompanies this distribution
+# and is available at https://www.apache.org/licenses/LICENSE-2.0.
+#
+# This Source Code may also be made available under the following Secondary
+# Licenses when the conditions for such availability set forth in the
+# Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+# version 2 with the GNU Classpath Exception [1] and GNU General Public
+# License, version 2 with the OpenJDK Assembly Exception [2].
+#
+# [1] https://www.gnu.org/software/classpath/license.html
+# [2] http://openjdk.java.net/legal/assembly-exception.html
+#
+# SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+###############################################################################
+
+if(OMR_ENV_DATA64)
+	list(APPEND OMR_PLATFORM_DEFINITIONS
+		-D_RISCV64_
+	)
+else()
+	list(APPEND OMR_PLATFORM_DEFINITIONS
+		-D_RISCV32_
+	)
+endif()
+
+# Testarossa build variables. Longer term the distinction between TR and the rest
+# of the OMR code should be heavily reduced. In the mean time, we keep
+# the distinction
+set(TR_HOST_ARCH r)
```

Comment:
I realize there is some precedent with 3 of the other architectures to use a much abbreviated form to describe the host architecture.  I admit I'm not particularly keen on the brevity.  As you're adding support for a new architecture, I think using `riscv` makes this more readable.  Also, it makes it more consistent with the subarchitectures defined below: `riscv32` and `riscv64`.

Reply:
Plain `r` is what @shingarov originally used so I kept it. Also I must admit I like just `r` more, but maybe it's just me. For now I kept `r`, but if there's a strong preference to `riscv` I'll rename. @shingarov? @mstoodle?

Reply:
I don't recall which medium it was in (I think it was in one of the compiler architecture calls but it might be somewhere else) but I definitely remember we did discuss this name issue around April or so.  Both @janvrany and I like `r`, and no one felt really against it, therefore we decided to keep `r` for now.  Frankly, I use the PPC port for reference on what "should" be implemented, so I do `cd .../p/...` and `cd .../r/...` all the time and now this has become a finger habit (let alone the command history) so to change it would be slightly disruptive to me.  But then again: if someone comes up with a strong argument why we should name it something else, I am not going to be up in arms.

Reply:
One argument "against" using `r` as the directory name is that there is a language named "R" which could be confusing (although that directory name in the path is always a platform rather than a language).

I confess, I do not naturally associate `r` with "RISC-V" . Is this a short form that people in the community typically us when referring to the platform?